docs/source/parameter_handling.rst

```diff
@@ -39,4 +39,51 @@ The parameters can then be accessed as
     [('param1', 7.0), ('param2', 4.0)]
     >>> p.exists("param1")
     True
+    
+    
+.. code-block:: python
+ 
+    # print all data inside Param
+    def printAll(x):
+      if (x.size()):
+        for i in x.keys():
+          print("Key: ", i, "Value: ", x[i])
+
+      else :
+          print("no data availabe")
+
+
+    new_p = Param()
+    if (p.empty() == False): #check p is not empty
+      new_p = p               #new soft copy of p generate with name "p"
+
+    #adding 3 more keys with different names
+    new_p.setValue("param2", 9.0, "This is value 9")
+    new_p.setValue("example1", 6.0, "This is value 6")
+    new_p.setValue("example2", 8.0, "This is value 8")
+    new_p.setValue("example3", 10.0, "This is value 10")
+
+    #new_p merge in p, p will update same key entries
+    p.merge(new_p)
+    print(" print All inside p after merge new_p ")
+    printAll(p) #print all data 
+    
+    
+
+.. code-block:: python
+ 
+    print(" print All inside new_p ")
+    printAll(new_p) #print all data
+
+    new_p.remove("example3")
+    print(" print All inside new_p  after remove function ")
+    printAll(new_p) #print all data
+
+    new_p.removeAll("exam")
+    print(" print All inside new_p  after remove all with prefix exam ")
+    printAll(new_p) #print all data
 
+    if (p == new_p):
+      new_p.clear() #clear all entries
+    print(" print All inside new_p  after clear function ")
```

Comment:
replace double spaces.
maybe rephrase in a less technical manner: print("print Param object after we cleared it.")